src/test/ui/rfc-2011-nicer-assert-messages/feature-gate-generic_assert.rs

```diff
@@ -0,0 +1,41 @@
+// compile-flags: --test
+// run-pass
+
+#![feature(generic_assert, generic_assert_internals)]
+
+#[derive(PartialEq)]
+struct NoDebug(i32);
+
+#[should_panic(expected =
+r#"Assertion failed: &debug == &3
+With expansion: &1 == &3"#
+)]
+#[test]
+fn debug_by_value_addr() {
+    let debug = 1;
+    assert!(&debug == &3);
+}
+
+#[should_panic(expected =
+r#"Assertion failed: debug == 3
+With expansion: 1 == 3"#
+)]
+#[test]
+fn debug_by_value_binary() {
+    let debug = 1;
+    assert!(debug == 3);
+}
+
+
+#[should_panic(expected =
+r#"Assertion failed: !debug
```

Comment:
```suggestion
r"Assertion failed: !debug
```
`#`s are not required if you are not usihng `"`s inside the string.